### PR TITLE
fix ha filter test is not stable

### LINKED ISSUE #2672: TestHAFilter is not stable
## Bug Report

https://github.com/pingcap/tidb-operator/pull/2664/checks?check_run_id=753184840
Raw logs: https://pipelines.actions.githubusercontent.com/XmlGqt0jpH5ovEMHxNC2UKP0AhbomTmki7mJypwpILn8wVQr4g/_apis/pipelines/1/runs/3669/signedlogcontent/5?urlExpires=2020-06-09T09%3A33%3A19.7032938Z&urlSigningMethod=HMACV1&urlSignature=lW%2FS78LIoHyEs119%2FeAaZzsx4KdaylYB%2BW2Q9Cg4vbg%3D

```
##[error]    ha_test.go:444: three topologies, two pods scheduled on these two topologies, replicas is 5, return two topologies
    ha_test.go:800: 
        Expected
            <[]string | len:2, cap:2>: ["zone2", "zone1"]
        to equal
            <[]string | len:2, cap:2>: ["zone1", "zone2"
```

This test is not stable. Maybe we need to sort before comparing? cc @PengJi can you take a look?

### PR DESCRIPTION
**Pull request by PengJi**
### What problem does this PR solve? 
fix #2672

### What is changed and how does it work?
Sort before comparing zones

### Check List 

Tests 

 - Unit test

Code changes

 - Has Go code change

Side effects

 - None

Related changes

 - Need to cherry-pick to the release branch

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: pkg/scheduler/predicates/ha_test.go [hunk 1]
```diff
@@ -1182,6 +1182,15 @@ func getTopologies(nodes []apiv1.Node, topologyKey string) []string {
 	return arr
 }
 
+func getSortedTopologies(nodes []apiv1.Node, topologyKey string) []string {
+	arr := make([]string, 0)
+	for _, node := range nodes {
+		arr = append(arr, node.Labels[topologyKey])
+	}
```
Comment (cofyc) [inline pkg/scheduler/predicates/ha_test.go:1189]: reuse `getTopologies` or remove `getTopologies`?

Comment (PengJi) [reply pkg/scheduler/predicates/ha_test.go:1189]: remove `getTopologies `, replace `getTopologies ` with `getSortedTopologies`？

Comment (cofyc) [reply pkg/scheduler/predicates/ha_test.go:1189]: I think either is fine

**Pull request merged**